cmake/scripts/ROOTConfig.cmake.in

```diff
@@ -108,6 +108,9 @@ if(ROOT_minuit2_omp_FOUND)
   find_dependency(Threads)
 endif()
 if(ROOT_vdt_FOUND AND NOT TARGET VDT::VDT)
+  if(ROOT_builtin_vdt_FOUND)
+    set(Vdt_ROOT "${_thisdir}/.." CACHE PATH "Location of (the builtin) VDT")
```

Comment:
on line 102, we could add sth along this line: `else() set (nlohmann_json... "${_thisdir}/.." CACHE PATH "Location of (the builtin) nlohmann-json")`

Reply:
I do not think that we need special rule here for builtin `nlohmann/json.hpp`.
Only when external `nlohmann_json` is used, it linked as `PUBLIC` dependency and need to be there when 
correspondent library linked by user application. This is how it defined in REve:
```
if(builtin_nlohmannjson)
   target_include_directories(ROOTEve PRIVATE ${CMAKE_SOURCE_DIR}/builtins)
else()
   target_link_libraries(ROOTEve PUBLIC nlohmann_json::nlohmann_json)
endif()
```
So to say - builtin version should not be seen by user cmake.

Reply:
Hi @linev, but find_package (ROOT) fails because it does not find where nlohmann json is, even if I do not need nlohman_json at all in my software.

Reply:
I just want to understand that kind of changes you want to have for `nlohmann_json`?

Reply:
😕 Have a look at line 100-102. The logic is actually the opposite for nlohmann_json. CMake **only** calls `find_dependency(nlohmann_json...)` if it's **not** built-in. For a built-in nlohmann_json the CMake setup does nothing. As it expects that the normal ROOT include path would make the `nlohmann/json.hpp` file visible as well. 🤔

But this also means that apparently the binary downloaded by @ferdymercury does **not** have nlohmann_json included in it. If it did, ROOT would not be looking for it.

Long story short, I didn't think that we needed to do anything with the JSON dependency. Also because that's not something that would've changed recently. I only modified how VDT is handled in 6.30. The JSON dependency did not change wrt. 6.28 as far as I can see. 🤔

Reply:
Hmm... I myself would re-write the JSON dependency handling in the same way as VDT is handled now... But that we should do in a separate PR.

And yes, it very much seems that in the 6.30 binary `builtin_nlohmannjson` was turned off for some reason. 🤔

Reply:
> I just want to understand that kind of changes you want to have for `nlohmann_json`?

Hmmm, I would like to use `find_package(ROOT)` with binary release 6.30.02 without forcing me to install nlohmann_json externally.

Reply:
> 
> And yes, it very much seems that in the 6.30 binary `builtin_nlohmannjson` was turned off for some reason. 🤔

Oh, I see. The same happened with builtin_afterimage. If this was done on purpose, then we should add nlohmann_json to the prerequisites, as I did with afterimage: https://github.com/root-project/web/pull/936/files